x/concentrated-liquidity/math/math.go

```diff
@@ -132,11 +170,12 @@ func GetNextSqrtPriceFromAmount0InRoundingUp(sqrtPriceCurrent, liquidity, amount
 		return sqrtPriceCurrent
 	}
 
-	product := amountZeroRemainingIn.Mul(sqrtPriceCurrent)
+	// MulTruncate to get a smaller denominator
+	product := amountZeroRemainingIn.MulTruncate(sqrtPriceCurrent)
 	// denominator = product + liquidity
 	denominator := product
 	denominator.AddMut(liquidity)
-	return liquidity.Mul(sqrtPriceCurrent).QuoRoundupMut(denominator)
+	return liquidity.MulRoundUp(sqrtPriceCurrent).QuoRoundupMut(denominator)
```

Comment:
Note: the reason why this change was made is to ensure that if any rounding error occurs, the value computed by this function is always greater than the true value per requirements. With the old change, banker's rounding can break this guarantee.

No test ended up breaking from this change.